src/sentry/api/endpoints/organization_discover.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import absolute_import
+
+from datetime import datetime
+
+from rest_framework.response import Response
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationDiscoverPermission
+from sentry.models import Project, ProjectStatus
+
+from sentry.utils import snuba
+
+from django.core.exceptions import ValidationError
+
+
+class OrganizationDiscoverEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationDiscoverPermission, )
+
+    def do_query(self, start, end, selected_columns=[], conditions=[], having=[], filters={},
+                 aggregations=[], orderby='-last_seen', limit=None, groupby=[], rollup=None):
+
+        snuba_results = snuba.raw_query(
+            start=start,
+            end=end,
+            groupby=groupby,
+            selected_columns=selected_columns,
+            conditions=conditions,
+            having=having,
+            filter_keys=filters,
+            aggregations=aggregations,
+            orderby=orderby,
+            limit=limit,
+            rollup=rollup,
+            referrer='discover',
+        )
+
+        return snuba_results
+
+    def post(self, request, organization):
+        data = request.DATA
+
+        filters = {
+            'project_id': data['projects']
+        } if 'projects' in data else None
+
+        selected_columns = data['fields'] if 'fields' in data else None
```

Comment:
i think you can just shorten this to `selected_columns = data.get('fields')` (and same for all of the ones below)